packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -1212,42 +1303,27 @@ class _ClassOrInstanceCountColumn extends ColumnData<Reference> {
     if (dataObject.name == MemoryController.libraryRootNode ||
         dataObject.name == MemoryController.classRootNode) return '';
 
-    if (dataObject.isExternals) {
-      if (dataObject.hasCount) return dataObject.count;
-
-      var count = 0;
-      for (ExternalReference externalRef in dataObject.children) {
-        count += externalRef.children.length;
-      }
-      return count;
-    } else if (dataObject.isExternal) {
+    if (dataObject.isExternal) {
       return dataObject.children.length;
-    } else if (dataObject.isFiltered) {
-      int sum = 0;
-      final FilteredReference filteredRef = dataObject;
-      for (final LibraryReference child in filteredRef.children) {
-        for (final HeapGraphClassLive liveClass in child.actualClasses) {
-          // Have the instances been realized (null implies no)
-          if (liveClass.instancesCount != null) {
-            sum += liveClass.instancesCount;
-          }
-        }
-      }
-
-      return sum;
     } else if (dataObject.isAnalysis && dataObject is AnalysisReference) {
       final AnalysisReference analysisReference = dataObject;
       final count = analysisReference.countNote;
       return count == null ? '' : count;
+    } else if (dataObject.isSnapshot && dataObject is SnapshotReference) {
+      final SnapshotReference snapshotRef = dataObject;
+      int count = 0;
+      for (final child in snapshotRef.children) {
+        count += _computeCount(child);
+      }
+      return count == null ? '--' : '$count';
     }
 
     final count = _computeCount(dataObject);
-
     return count == null ? '--' : count;
   }
 
   /// Return of null implies count can't be computed.
-  int _computeCount(Reference ref) {
+  dynamic _computeCount(Reference ref) {
```

Comment:
comment seems wrong. this is really returning a String now (at least in some cases) and null isn't returned as `null` is converted to `--`.

Reply:
/// Return a count based on the Reference type e.g., library, filtered,
  /// class, externals, etc.  Only compute once, store in the Reference.
  int _computeCount(Reference ref) {